### PR TITLE
feat(Auth): Adding UpdateUserAttribute API implementation

### PR DESCRIPTION
**Pull request by harsh62**
*Description of changes:*
The purpose of the PR is to add implementation of Update User Attribute API. 

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
~~- [ ] Documentation update for the change if required~~
- [x] PR title conforms to conventional commit style
~~- [ ] If breaking change, documentation/changelog update with migration instructions~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthUpdateUserAttributeOperation.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                                                  clientMetadata: clientMetaData,
+                                                  userAttributes: attributeList.map({ $0.sdkClientAttributeType() }))
+            
+            let result = try await userPoolService.updateUserAttributes(input: input)
+            
+            if self.isCancelled {
+                return
+            }
+            
+            guard let codeDeliveryDetailsList = result.codeDeliveryDetailsList else {
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthUpdateUserAttributeOperation.swift:85]: Does the service always return this value? What about cases where we do not need to verify the update?

File: AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/AuthUserAttributesTests/AuthUserAttributesTests.swift [hunk 2]
```diff
@@ -100,6 +100,23 @@ class AuthUserAttributesTests: AWSAuthBaseTest {
...
         }
         wait(for: [updateExpectation], timeout: networkTimeout)
+        
+        let fetchUserAttributeExpectation = expectation(description: "Fetch User Attribute operation should complete")
+        
+        _ = Amplify.Auth.fetchUserAttributes(listener: { result in
+            switch result {
+            case .success(let attributes):
+                if let emailAttribute = attributes.filter({ $0.key == .email }).first {
+                    XCTAssertEqual(emailAttribute.value, updatedEmail)
```
Comment (royjit) [inline AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/AuthUserAttributesTests/AuthUserAttributesTests.swift:110]: Does this work? We did not verify the attribute correct?

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthUpdateUserAttributeOperation.swift (hunk 1 above)
Comment (harsh62) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/UserOperations/AWSAuthUpdateUserAttributeOperation.swift:85]: So  the list is an optional that's why I need to place a guard around it. 
```
public var codeDeliveryDetailsList: [CognitoIdentityProviderClientTypes.CodeDeliveryDetailsType]?
```

In cases where the next step is not needed, the list would be empty and not `nil`. In those scenarios, the for loop underneath should handle it. Which is as follows. It iterates the dictionary and if the attribute doesn't exist, will update the result accordingly. 

```
            // Check if all items are added to the dictionary
            for item in attributeList where finalResult[item.key] == nil {
                let updateAttributeResult = AuthUpdateAttributeResult(isUpdated: true, nextStep: .done)
                finalResult[item.key] = updateAttributeResult
            }
```

File: AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/AuthUserAttributesTests/AuthUserAttributesTests.swift (hunk 2 above)
Comment (harsh62) [reply AmplifyPlugins/Auth/Tests/AuthHostApp/AuthIntegrationTests/AuthUserAttributesTests/AuthUserAttributesTests.swift:110]: Yes. So if you check the entire test `func testSuccessfulUpdateEmailAttribute() throws`.
* it first sign up and sign in the user. 
* Updates the email 
* Fetches attributes to check if email was updated. (Added this additional step to reinforce the test)

**Review by royjit — APPROVED** [review]

**Review by royjit — APPROVED** [review]

**Pull request merged**